js/runtime.go

```diff
@@ -123,22 +123,44 @@ func (r *Runtime) extractOptions(exports otto.Value, opts *lib.Options) error {
 	return nil
 }
 
-func (r *Runtime) loadFile(filename string) (otto.Value, error) {
-	// To protect against directory traversal, prevent loading of files outside the root (pwd) dir
-	path, err := filepath.Abs(filename)
+func (r *Runtime) loadJSCode(src *lib.SourceData) (otto.Value, error) {
```

Comment:
Thought: could we just merge this into `load()` and have `loadFile()` produce a SourceData blob?